src/child/index.js

```diff
@@ -20,13 +21,22 @@ require('script!../../node_modules/BitFlux/dist/bitflux.js');
 /* eslint-enable import/no-unresolved */
 
 fin.desktop.main(() => {
+    const store = fin.desktop.Window.getCurrent().contentWindow.opener.store;
+    store.dispatch(open());
+    const windowState = indexSelector(store.getState());
+
     fin.desktop.Window.getCurrent().contentWindow.addEventListener('beforeunload', () => {
         fin.desktop.Window.getCurrent().contentWindow.opener.store.dispatch(close());
         unmountComponentAtNode(document.getElementById('app'));
     });
 
-    const store = fin.desktop.Window.getCurrent().contentWindow.opener.store;
-    store.dispatch(open());
+    // Ensure that the window is started in the same
+    // state as the previous session, if available
+    if (windowState.isCompact) {
```

Comment:
Did you consider doing this before the window is created and shown?

When creating the window, the previous window state (rather than the default) could be set in the [options object](http://cdn.openfin.co/jsdocs/stable/tutorial-window.constructor.html) passed to the window constructor; if anything can't be set in this option object and needs to be done (e.g. a method call), then I think this should be done before the window is shown.